### PR TITLE
Add HW3 tests for Muwen320 (smoke/one-shot/edge/pattern)

### PR DESCRIPTION
**Pull request by Muwen320**
This PR adds the four required tests for Homework 3:
	•	Smoke test: checks that calculate_github_score runs successfully on a basic project dictionary.
	•	One-shot test: verifies normalize(100, 100) returns 1.0.
	•	Edge test: ensures projects with many open issues do not receive a higher score than those with few issues.
	•	Pattern test: confirms that age_factor increases (or stays the same) as project age increases.

### CODE REVIEW

File: hw3-test/test_hw3_muwen320.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        "open_issues": 5,
+        "created_at": created,
+        "pushed_at": pushed,
+    }
+
+    score = calculate_github_score(project, max_stars=1000, max_forks=500, max_watchers=200)
+
+    assert isinstance(score, (int, float))
+    assert score >= 0
+
```
Comment (Wenshu0206) [inline hw3-test/test_hw3_muwen320.py:54]: The test is clear, well-structured, and correctly validates the basic smoke-test behavior of calculate_github_score(). It constructs a realistic project dictionary with properly formatted timestamps and checks both type safety and non-negativity of the returned score, which aligns with the expectations of a smoke test. The use of dynamic timestamps (now, created, pushed) avoids brittleness and makes the test stable over time. Overall, the test is simple, readable, and effectively confirms that the function runs without errors on valid input.

File: hw3-test/test_hw3_muwen320.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    reviewer: Wenshu0206
+    category: one-shot test
+
+    One-shot test:
+    Fixed input–output pair. When value == max_value, normalization
+    should return exactly 1.0.
+    """
+    result = normalize(100, 100)
+    assert result == 1.0, f"Expected 1.0 but got {result}"
+
```
Comment (Wenshu0206) [inline hw3-test/test_hw3_muwen320.py:68]: This one-shot test is straightforward and effectively verifies a key boundary condition of the normalize() function. The intent is clearly documented, and the assertion directly checks that a full-value input produces an exact normalized output of 1.0. The test is minimal, readable, and correctly captures the expected behavior for this edge case.

**Review by Wenshu0206 — DISMISSED** [review]

File: hw3-test/test_hw3_muwen320.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    project_few_issues = {**base, "open_issues": 1}
+    project_many_issues = {**base, "open_issues": 5000}
+
+    score_few = calculate_github_score(project_few_issues, 10_000, 5_000, 2_000)
+    score_many = calculate_github_score(project_many_issues, 10_000, 5_000, 2_000)
+
+    assert score_many <= score_few, (
+        f"Expected score with many issues <= score with few issues, "
+        f"got many={score_many}, few={score_few}"
+    )
```
Comment (Chase-Zou) [inline hw3-test/test_hw3_muwen320.py]: The test only checks ordering (<=). If the score ignores open_issues, this test still passes. Recommend tightening to score_many < score_few or adding a check that the difference is non-zero to ensure the function actually reacts to issue count.

File: hw3-test/test_hw3_muwen320.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    mid_score = age_factor(created_mid)
+    old_score = age_factor(created_old)
+
+    for s in (new_score, mid_score, old_score):
+        assert 0.0 <= s <= 1.0
+
+    assert new_score <= mid_score <= old_score, (
+        f"Expected new <= mid <= old, got "
+        f"new={new_score}, mid={mid_score}, old={old_score}"
+    )
```
Comment (Chase-Zou) [inline hw3-test/test_hw3_muwen320.py]: Using only three timestamps is a weak pattern test. Consider adding 1–2 more intermediate ages or random samples to better verify the monotonic trend instead of relying on just new/mid/old.

**Review by Chase-Zou — APPROVED** [review]
